### DISCUSSION TITLE
Set Order for a test case at run time

### DISCUSSION THREAD

**Discussion by ayankumaar** [nunit/nunit#3864] [General]
Is there any way to set the "Order" attribute for a test case at run time from C# code ?

**Comment by mikkelbu** [comment]
The effect of the `OrderAttribute` is to set an `Order` property on the test or the fixture. And the order is then used when the tests are executed, so you can do something like

(usually, the testcases from a source are run in order, but since we have added an explicit order we run in the following order: "TestB", "TestA", and "TestC").
```csharp
        public static IEnumerable<TestCaseData> GetCases()
        {
            yield return new TestCaseData("TestA").SetProperty("Order", 10);
            yield return new TestCaseData("TestB").SetProperty("Order", 1);
            yield return new TestCaseData("TestC");
        }


        [TestCaseSource(nameof(GetCases))]
        public void Test1(string data)
        {
            Thread.Sleep(5000);
            Assert.Pass();
        }
```

**Reply by ayankumaar** [reply]
@mikkelbu thanks for the update. But, my requirement here is something different. I am reading the tests from the dll (the binary which contains the tests) using reflection in a different exe. And after reading the tests from the dll I am passing it to the dotnet CLI for execution. 

Now, after reading it using reflection in a separate exe, how can I set the order at run time before passing the tests to the dotnet CLI ?

**Reply by mikkelbu** [reply]
Exactly how are you passing the tests to the dotnet CLI?

**Reply by ayankumaar** [reply]
@mikkelbu I am running the tests like this:

dotnet test --filter "FullyQualifiedName=TestScripts.Tests.SmokeTest.TestA"|"FullyQualifiedName=TestScripts.Tests.SmokeTest.TestB"|"FullyQualifiedName=TestScripts.Tests.SmokeTest.TestC"   "MyTests.dll"

Here my expectation is, I should be able to set the order at run time. So that it runs at the specified order i.e. TestA, TestB, TestC

**Reply by mikkelbu** [reply]
@ayankumaar I'm sorry, but it is not possible to specify the order like that. The filter says "I would like to run a test with name "TestScripts.Tests.SmokeTest.TestA" or with name "TestScripts.Tests.SmokeTest.TestB" or with name "TestScripts.Tests.SmokeTest.TestC"", but it does not say in which order. So the filter just specifies a set of test to run.

Note that the filter could just as well specify run all tests that have "UnitTest1" in their FullyQualifiedName, i.e. `--filter "FullyQualifiedName~UnitTest1"`.

**Reply by ayankumaar** [reply]
@mikkelbu thanks for the reply. But, any way to read the dll using reflection and setting the order and then running something like: dotnet test --filter "FullyQualifiedName~UnitTest1" ?

**Reply by mikkelbu** [reply]
@ayankumaar Not as far as I know. The only way to do something similar to this is to run several `dotnet test` for sequentially - one for each test. This will of course have some overhead in starting the test process and also generate several result files, but I think this is the only solution.
